sdk/src/main/java/com/bugsnag/android/Client.java

```diff
@@ -206,12 +207,23 @@ public void run() {
 
         config.addObserver(this);
 
-        // Flush any on-disk errors
-        errorStore.flushOnLaunch(errorReportApiClient);
-
         boolean isNotProduction = !AppData.RELEASE_STAGE_PRODUCTION.equals(
             AppData.guessReleaseStage(appContext));
         Logger.setEnabled(isNotProduction);
+
+        // Flush any on-disk errors
+        Async.run(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    Thread.sleep(10); // allow users to set custom API clients
+                    errorStore.flushOnLaunch(errorReportApiClient);
+                } catch (InterruptedException exception) {
+                    Logger.warn("Failed to flush errors on launch", exception);
+                }
+            }
+        });
+        clientInitialised = true;
```

Comment:
While this fixes the immediate issue, the longer term problem to solve here is that some configuration options should be set correctly when the client is created. Properties like `errorReportingApiClient` should be on `Configuration` so that the client can then immediately use what has been configured if its available, and then we should phase out the convenience setters for those properties in the `Bugsnag` interface.

Adding a delay here increases the risk of reports not being sent if the app enters a crash loop.